src/Splunk.Client/Splunk/Client/JobCollection.cs

```diff
@@ -234,12 +259,27 @@ async Task<Job> CreateAsync(IEnumerable<Argument> arguments, DispatchState requi
 
             Job job = new Job(this.Context, this.Namespace, name: searchId);
 
-            await job.GetAsync().ConfigureAwait(false);
+            await job.GetAsync(requiredState).ConfigureAwait(false);
```

Comment:
Previously in `CreateAsync()` we would ignore the `requiredState` value and by default wait for every job to reach a created state before returning - resulting in timeouts.

Reply:
This version will still time out if required state is running and we get queued though?